util/include/digest_creator.hpp

```diff
@@ -0,0 +1,30 @@
+// Concord
+//
+// Copyright (c) 2022 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").  You may not use this product except in
+// compliance with the Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright notices and license terms. Your use of
+// these subcomponents is subject to the terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#pragma once
+
+namespace concord::util::digest {
+
+// It is responsible for generating the digest.
+class DigestCreator {
```

Comment:
I think an interface `DigestCreator` class is not needed, since there's no factory which can return different types of creators, but rather is defined at compile time by `using DigestGenerator = OpenSSLDigestCreator<SHA2_256>;`
If a specific generator won't implement the required methods, there will be a compilation error.
This also prevents from using a static `compute` function as previously and demands a DigestGenerator class instantiation before use.
Also there may be a confusion between `DigestCreator` and `DigestGenerator`

Reply:
Thank you for the review comment.

The idea of implementing an interface is because we wanted all `cryptopp`, `openssl`, and other future libraries (if included) to implement and follow the same methods for newly added library(s) to be inclined with the existing code with less or no changes in the existing codebase.
Would also help in A/B testing.

Removing static property of `compute()` function is done for ease of multi-threading.

Yes, there is no factory. We may implement factory if required later too, so keeping interface is a correct thing to do if I am not wrong.

Reply:
I don't see how my comments contradict what you wanted to achieve

Reply:
Addressed: https://github.com/arc-vmware/concord-bft/pull/5